File: Golang/Contains-Duplicate.go
/*
  1st Solution:
    Time: O(n log(n))
    Space: O(1)
*/
func containsDuplicate(nums []int) bool {
    sort.Slice(nums, func(i, j int) bool {
        return nums[i]<nums[j]
    })
    
    for i := 0; i<len(nums)-1; i++ {
        if nums[i] == nums[i+1] {
            return true
        }
    }
    return false
}

//-------------------------------------------------

/*
  2nd Solution:
    Time: O(n)
    Space: O(n)
*/
func containsDuplicate(nums []int) bool {
    visited := map[int]int{}
    
    for i := 0; i<len(nums); i++ {
        if visited[nums[i]] == 0 {
            visited[nums[i]]+=1
        } else {
            return true
        }
    }
    return false
}

File: Study Plans/Data Structure/Data Structure 1/Java/Contains-Duplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        
        Set<Integer> visited = new HashSet<>();
        
        for (int i = 0; i<nums.length; i++) {
            visited.add(nums[i]);
        }
        
        return visited.size() < nums.length;
    }
}
